src/Core/src/Core/IVisualDiagnosticsOverlay.cs

```diff
@@ -0,0 +1,127 @@
+		/// <param name="scrollToView">When adding the adorner, scroll to the element. Only applies if the element is contained in an <see cref="IScrollView"/>.</param>
+		bool AddAdorner(IAdorner adorner, bool scrollToView);
+
+		/// <summary>
+		/// Adds a new adorner to the Visual Diagnostics Overlay. Uses the default Adorner for drawing.
+		/// </summary>
+		/// <param name="visualElement"><see cref="IVisualTreeElement"/>.</param>
+		/// <param name="scrollToView">When adding the adorner, scroll to the element. Only applies if the element is contained in an <see cref="IScrollView"/>.</param>
+		bool AddAdorner(IVisualTreeElement visualElement, bool scrollToView);
+
+		/// <summary>
+		/// Removes adorner from Visual Diagnostics Overlay.
+		/// </summary>
+		/// <param name="adorner"><see cref="IAdorner"/>.</param>
+		bool RemoveAdorner(IAdorner adorner);
+
+		/// <summary>
+		/// Removes all adorners from the Visual Diagnostics Overlay.
+		/// </summary>
+		void RemoveAdorners();
+
+		/// <summary>
+		/// Removes all adorners containing the inner <see cref="IVisualTreeElement"/>.
+		/// </summary>
+		/// <param name="visualElement"><see cref="IVisualTreeElement"/>.</param>
+		bool RemoveAdorners(IVisualTreeElement visualElement);
+
+		/// <summary>
+		/// Adds scrollable element handles attached to <see cref="IScrollView"/>.
+		/// Used for tracking when a user has scrolled, in order to update the layer to redraw.
+		/// </summary>
+		void AddScrollableElementHandlers();
+
+		/// <summary>
+		/// Adds scrollable element handle attached to <see cref="IScrollView"/>.
+		/// Used for tracking when a user has scrolled, in order to update the layer to redraw.
+		/// </summary>
+		void AddScrollableElementHandler(IScrollView view);
+
+		/// <summary>
+		/// Removes any existing scrollable element handles attached to <see cref="IScrollView"/>.
+		/// </summary>
+		void RemoveScrollableElementHandler();
+
+		/// <summary>
+		/// Automatically scroll to a given element within the view, if available.
+		/// </summary>
+		/// <param name="element">Element to scroll to.</param>
+		void ScrollToView(IVisualTreeElement element);
+
+#if ANDROID
+		/// <summary>
+		/// Gets the Scroll Views in a given window, to be handled by the layer for
+		/// when they scroll to update the underlying adorners.
+		/// </summary>
+		IReadOnlyDictionary<IScrollView, Android.Views.View> ScrollViews { get; }
+#elif IOS
+		/// <summary>
+		/// Gets the Scroll Views in a given window, to be handled by the layer for
+		/// when they scroll to update the underlying adorners.
+		/// </summary>
+		IReadOnlyDictionary<IScrollView, IDisposable> ScrollViews { get; }
+#elif WINDOWS
+		/// <summary>
+		/// Gets the Scroll Views in a given window, to be handled by the layer for
+		/// when they scroll to update the underlying adorners.
+		/// </summary>
+		IReadOnlyDictionary<IScrollView, Microsoft.UI.Xaml.Controls.ScrollViewer> ScrollViews { get; }
```

Comment:
For things like this, could we move the native things to a `using NativeScrollView = Microsoft.UI.Xaml.Controls.ScrollViewer` so that we can avoid any potential issues with names, types or any copy-paste issues.